### DISCUSSION TITLE
Discord community call: 28 September 2024

### DISCUSSION THREAD

**Discussion by BenJamesBen** [freesewing/freesewing#7029] [Support] [answered]
You are all invited to our Discord hangout.
Please leave a comment here or [on Discord](https://discord.freesewing.org/) to add things to the agenda.
This page will be updated after the call with notes.

### About

FreeSewing's Discord hangout is an informal hangout/chat/meeting that is open to everyone.
The call is organized/hosted by @tangerineshark.
The call does not have a fixed structure or agenda items. Everyone is free to suggest agenda items, ask questions, or bring ideas.

### Details

Location: https://discord.freesewing.org/ (meeting room: 🔊voice chat)
Date: September 14th, 2024 4pm UTC, which can be found in your local time zone [here](https://savvytime.com/converter/utc-to-belgium-antwerpen-ca-los-angeles-il-chicago-ny-new-york-city-brazil-sao-paulo-united-kingdom-london-france-paris-south-africa-johannesburg-msd-india-mumbai-hkt-japan-tokyo-australia-sydney/sep-28-2024/4pm).
Or check the Events tab in Discord!
The meetings will always be at 6pm, Antwerp time.

### Agenda

- TBD

**Comment by joostdecock** [comment]
Hey y'all.
I won't be in the call tonight. I am traveling to attend a work-related event tonight.

I was planning to -- and to some extend still am -- using this travel time to work on FreeSewing. But last night, I had a bit of a mental breakdown because I suddenly felt absolutely overwhelmed by all the work that is sitting in my inbox.

It's arguably not new, and I don't think anyone is waiting for yet another open source maintainer apologizing for being busy.
However, last night was the first time that the feeling crossed over from _phew, this is a lot_ into _I can't do this any longer_.

There's a lot that's been going on in the last couple of months (in my life that is), so my frazzled state of mind is probably not even related to FreeSewing, but I do think I need to make some changes before my mental health gets any worse.

So, effective immediately, I will work towards making FreeSewing simpler to maintain, and put more strict boundaries and what we do and what we don't do.

This is a bit of a moving target, but the following changes will be implemented going forward:

- No more support for multiple languages. English only. All other languages will be removed, along with the complexity that comes with i18n.
- Documentation will be ported to Docusaurus, and we'll keep it simpler, using less custom components.
- Documentation will be revisited, we will keep only what is truly relevant to what FreeSewing does

I appreciate that not everybody may _like_ these changes.
Fear not, there is an alternative approach: I am happy to hand over FreeSewing to a new maintainer if there's a fitting candidate.

I will start working on these changes assuming that nobody is going to volunteer for this job. But if it turns out I'm wrong, do not hesitate to reach out.

**Reply by BenJamesBen** [reply]
@joostdecock Could we please discuss these changes before they are implemented? Perhaps at the next meeting? 

I think we need to discuss this so community members can better prepare themselves for the removal of i18n languages. And, to plan exit strategies and possible alternatives/workarounds.

**Reply by joostdecock** [reply]
Removing i18n from core would be a breaking change, so that's not something that would go into anything but a major release.

I'm mostly talking about i18n in our frontends, but even that will take time to implement.

There's also [the `prepurge` tag](https://github.com/freesewing/freesewing/tree/prepurge) which would be a good starting point for people looking to fork the status quo.

You can of course discuss this and as I've said, I'm happy to hear all proposals.

**Reply by HaasJona** [reply]
I unfortunately generally agree with removing i18n. It's hard to maintain and a lot of work and currently at least the German translation, which is the one  I understand) is pretty useless anyway as it's super incomplete and full of bad and confusing machine translations. 

And translating all the sewing instructions and so on doesn't really make sense if your time would be better invested in completing the English ones, which are also sometimes incomplete and sometimes unclear.

**Comment by BenJamesBen** [comment, marked as answer]
### September 28, 2024 Community Call

#### Notes:
- Just bought a new sewing machine 30 minutes ago!
- Joost wants to downsize FreeSewing to make it easier to maintain. 
- Removing i18n languages.
  - Can we change the UI/infrastructure to support translations that would be provided by and maintained by other organizations? 
  - Freeze it and provide an opportunity for others to pick up?
  - Can we streamline technical side of translations to make it easier to maintain?
  - Dutch language speakers are already used to using apps in English and using tools to work around this.
  - There are many users worldwide who don't speak English and will find it extremely difficult if there are no translations.
  - We don't need to have answers now, but we do need to freeze things as they currently are to give us time and the ability to resolve the issue.
  - There is a feeling from some that i18n translations are very important and are what make FreeSewing a good project.
  - As pointed out in the GitHub discussion, the current German translations for FreeSewing are very incomplete and inaccurate.
  - Consensus was to "freeze" i18n work temporarily (don't do work to drop i18n, but also don't continue active work on it) to allow us to discuss the issue further, possible alternatives, etc.
- Moving documentation to Docusaurus:
  - We need more information on what this will entail before having an opinion on the issue.
  - Generally supportive of any way documentation is implemented technically, as long as documentation exists.
  - Would like to know more about how this change will help us maintain documentation and reduce effort.
  - Benjamin F. will investigate Docusaurus, how it works, how Joost envisions us using it, what the new workflow would be like, what we need to know. And, will present a short summary for the community.
- Removing documentation and keeping only things relevant to FreeSewing.
  - What existing documentation is this?
  - General support for the idea of dropping non-relevant documentation, but it isn't clear exactly what this is.
  - Suggestion that we similarly "freeze" changes to and work on documentation to give us time to discuss the issue.
  - We don't know the benefits and costs of documentation currently and with the proposed changes.
- How can we help Joost?
  - Perhaps we can ask Joost questions at the next meeting?
  - If Joost is overextended, perhaps it would be adding to his pain if we were to ask him to attend a meeting and have him discuss these things. Perhaps if we to instead find deputy maintainers to help him, then that would be better?
  - New maintainers/co-maintainers/deputy maintainers.
  - It might be better for Joost if we communicate with him via the GitHub Discussion. What things people are willing to help with. To ask what things they could help with.
- Are these problems we could resolve by throwing money at them?
- Who has access to maintain the FreeSewing YouTube channel? There's a video we might want to add. The suggestion was to ask TangerineShark.
- Has FreeSewing done anything to thank Joost for all the work he's done? Sending him a card, for example? Saber.007 will investigate what we might be able to do.
- Does FreeSewing documentation include a dictionary? A brief overview of FreeSewing documentation was provided, things on the org side and on the dev side.

#### Attendees:
- Benjamin F.
- bobgeorgethe3rd
- Curvalicia (a.k.a. Anna Puk)
- Natalia
- RockerKitten
- Saber.007
- Vili
- Wouter.vdub
